overlord/state/task.go

```diff
@@ -38,19 +38,20 @@ type progress struct {
 //
 // See Change for more details.
 type Task struct {
-	state     *State
-	id        string
-	kind      string
-	summary   string
-	status    Status
-	clean     bool
-	progress  *progress
-	data      customData
-	waitTasks []string
-	haltTasks []string
-	lanes     []int
-	log       []string
-	change    string
+	state        *State
+	id           string
+	kind         string
+	summary      string
+	status       Status
+	waitedStatus Status
```

Comment:
After seeing how this is used the name is a bit confusing to me, maybe better `statusAfterWait` or something similar? Also please add a comment in the definition.

Reply:
I prefer this as well, thanks for the suggestion Alfonso.

Reply:
I added some doc strings for this as suggested, to make sure the `waitedStatus` is properly documented. I'm a bit -0 on the renaming though.